local-cli/runIOS/runIOS.js

```diff
@@ -213,19 +213,22 @@ function buildProject(xcodeProject, udid, scheme, configuration = 'Debug', launc
         console.log(data.toString());
       }
     });
+    let error;
     buildProcess.stderr.on('data', function(data) {
-      console.error(data.toString());
+      error = data.toString();
+      console.error(error);
     });
     buildProcess.on('close', function(code) {
       if (xcpretty) {
         xcpretty.stdin.end();
       }
+      if (error) {
```

Comment:
> Out of curiosity, have you tried playing around with `code`? I missed it for the first time.

I missed it too, actually. But now when you mentioned it, I'll go and check if it contains zero exit code for successful builds and non-zero code for failed builds. If it does, it would be more clean to use it.

> Also, since we don't really do anything than just `if(error)`, maybe instead of doing `error = data.toString()` we simply store a variable `let hasError` or something.

I agree. If `code` argument contains what I suspect it to contain, then this additional variable may not even be needed. Otherwise, I'll change this code according to your suggestion.